src/components/sidebar/right-sidebar/RightSidebar.tsx

```diff
@@ -7,12 +8,113 @@ import "./RightSidebar.css"
 import Translate from "../../../utils/translate/translateService";
 import { Link } from "react-router-dom";
 
+import * as moment from 'moment';
+import 'react-dates/initialize';
+import { DateRangePicker, SingleDatePicker, DayPickerRangeController } from 'react-dates';
+import 'react-dates/lib/css/_datepicker.css';
+
 export default function RightSidebar() {
+  const [state, dispatch] = useContext(AppContext);
+
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+  const [focusedInput, setFocusedInput] = useState("startDate");
+
+  useEffect(() => {
+    if(state.airtable_records) {
+      const orderedDates = state.airtable_records
+      .filter(o => o.publish_date !== null && o.publish_date !== undefined)
+      .map(o => o.publish_date instanceof Array ? Date.parse(o.publish_date[0] as string) : Date.parse(o.publish_date as string))

```

Comment:
There might be some merge conflicts based on my rearchitecture changes, we are going to be using state.filteredRecords for now.